cocos/animation/core/pose.ts

```diff
@@ -0,0 +1,202 @@
+import { lerp } from '../../core';
+import { assertIsTrue } from '../../core/data/utils/asserts';
+import { Transform, __applyDeltaTransform, __calculateDeltaTransform } from './transform';
+import { TransformArray } from './transform-array';
+
+export class Pose {
+    readonly transforms: TransformArray;
+
+    readonly metaValues: Float64Array;
+
+    protected constructor (transforms: TransformArray, metaValues: Float64Array) {
+        this.transforms = transforms;
+        this.metaValues = metaValues;
+    }
```

Comment:
Q(@holycanvas ): What if we want to support non-skeletal animations in animation graph?

A(@shrinktofit ):
1. We have to add more value types into this `Pose` class: `Vec2`, `Vec3`.... These values should all be interpolate-able.

2. We must find a way so that we can know two clip "tracks" animates the same target so we can connect them together.